### PR TITLE
Added local-flavor support for Cuba

### PR DESCRIPTION
**Pull request by aaboffill**
**All Changes**

- [x] Add an entry to the docs/changelog.rst describing the change.

- [x] Add an entry for your name in the docs/authors.rst file if it's not
      already there.

- [x] Adjust your imports to a standard form by running this command:

      `isort --recursive --line-width 120 localflavor tests`

**New Fields Only**

- [X] Prefix the country code to all fields.

- [X] Field names should be easily understood by developers from the target
      localflavor country. This means that English translations are usually
      not the best name unless it's for something standard like postal code,
      tax / VAT ID etc.

- [x] Prefer '<country code>PostalCodeField' for postal codes as it's
      international English; ZipCode is a term specific to the United
      States postal system.

- [x] Add meaningful tests. 100% test coverage is not required but all
      validation edge cases should be covered.

- [x] Add `.. versionadded:: <next-version>` comment markers to new
      localflavors.

- [X] Add documentation for all fields.

### CODE REVIEW

**Comment by benkonrath** [comment]
Thanks for this pull request. You can rebase with master to fix the failing test. I made a mistake removing the Python 3.4 / master test config in Travis but it should be fixed now.

Let me know when this is ready for a review.

**Comment by aaboffill** [comment]
Hi! @benkonrath , thanks for your suggestion. I made the rebase and finally all checks have passed. This PR is ready for review. In few days i'll take some time to ensure 100% of the coverage of [localflavor/cu/forms.py](https://codecov.io/gh/django/django-localflavor/pull/292/diff).

**Comment by benkonrath** [comment]
I noticed that some of my commits are now included in this PR which probably isn't what you were intending. You should do a `git rebase` against the latest version from master to replay your commits on top of the latest version from master. Since you're rewriting the history of your branch, you'll have to force push your branch. While you do the rebase, you can also squash the commits into 1 commit if you want.

Here's a guide I found that might be helpful:
https://help.github.com/articles/about-git-rebase/

**Comment by aaboffill** [comment]
ohh! yes I see, thanks for the suggestion, I'll try ti fix that. Thanks for the suggestion @benkonrath ...

**Comment by aaboffill** [comment]
Hi! @benkonrath I organized the commits.

**Comment by benkonrath** [comment]
Sorry for the delay in getting to this. I'll be able to do a review early next week.

File: localflavor/cu/forms.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    .. versionadded:: 1.6
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a cuban region.'),
+    }
+
+    def clean(self, value):
+        super(CURegionField, self).clean(value)
+        if value in EMPTY_VALUES:
```
Comment (benkonrath) [inline localflavor/cu/forms.py]: EMPTY_VALUES -> self.empty_values

The same applies to `CUProvinceField`

File: localflavor/cu/forms.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+            return ''
+        try:
+            value = value.strip().lower()
+        except AttributeError:
```
Comment (benkonrath) [inline localflavor/cu/forms.py]: As far as I can see you don't need to catch an `AttributeError` here.

The same applies to `CUProvinceField`

File: localflavor/cu/forms.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+    default_error_messages = {
+        'invalid': _('Enter a valid zip code in the format XXXXX.'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(CUPostalCodeField, self).__init__(r'^[1-9]\d{4}$', *args, **kwargs)
+
+    def to_python(self, value):
+        value = super(CUPostalCodeField, self).to_python(value)
```
Comment (benkonrath) [inline localflavor/cu/forms.py]: You need to add `localflavor.compat.EmptyValueCompatMixin` to this class and add this python code:

```python
if value in self.empty_values:
    return value
```

This comes from a recent change that was made to properly support Django >= 1.11 (#298). You can search through the other localflavor modules to see examples of how to set this up if you're not sure.

File: localflavor/cu/forms.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            regex = re.compile(regex, re.UNICODE)
+        self._regex = regex
+        if hasattr(self, '_regex_validator') and self._regex_validator in self.validators:
+            self.validators.remove(self._regex_validator)
+        self._regex_validator = CUIdentityCardNumberValidator(regex=regex)
+        self.validators.append(self._regex_validator)
+
+    def to_python(self, value):
+        value = super(CUIdentityCardNumberField, self).to_python(value)
+        return value.strip()
```
Comment (benkonrath) [inline localflavor/cu/forms.py]: This also needs to use the `EmptyValueCompatMixin`as explained above.

File: localflavor/cu/forms.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a valid identity card number in the format XXXXXXXXXXX.'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(CUIdentityCardNumberField, self).__init__(r'^\d{11}$', *args, **kwargs)
+
+    def _set_regex(self, regex):
```
Comment (benkonrath) [inline localflavor/cu/forms.py]: You shouldn't override `_set_regex`. Either use `CharField` and set the `CUIdentityCardNumberValidator` using the `validators` arg in `__init__()` or use the `RegexField` like you do now and check the date in `clean()` after calling `super` instead of using the `CUIdentityCardNumberValidator`.

File: localflavor/cu/models.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        kwargs['max_length'] = 3
+        super(CUProvinceField, self).__init__(*args, **kwargs)
+
+    def deconstruct(self):
+        name, path, args, kwargs = super(CUProvinceField, self).deconstruct()
+        del kwargs['choices']
+        return name, path, args, kwargs
+
+
+class CUZipCodeField(CharField):
```
Comment (benkonrath) [inline localflavor/cu/models.py]: ZipCode -> Postal Code

The description also needs to be changed.

**Comment by aaboffill** [comment]
Hi @benkonrath , sorry for the delay. I update the code with all your suggestions. Thanks for your time :) , if you have another question/suggestion, please let me know.

**Comment by benkonrath** [comment]
@aaboffill Thanks for the update - it looks good. One some improvement that I missed last time: 'cuban' should be changed to 'Cuban' in the docstrings because it's a proper noun. Once that's updated, I'll merge this.

**Comment by aaboffill** [comment]
Hi! @benkonrath , thanks for your review :) ... I made your last suggestion related to `Cuban` instead of `cuban`

**Comment by benkonrath** [comment]
Merged via 328caadbe49e19108da2f8e5b679057db0a85f61.

Thanks!
